src/main/java/com/bandwidth/sqs/consumer/SqsConsumer.java

```diff
@@ -154,19 +156,27 @@ public int getNumPermits() {
      * Shuts down the consumer, and blocks until the shutdown completes with a
      * default timeout of 30 seconds
      */
-    public void shutdown() {
-        shutdown(DEFAULT_SHUTDOWN_TIMEOUT);
+    @PreDestroy
```

Comment:
this library is written not to assume it's running in any dependency injection environment; I think instead of putting this here, we should edit the client code where we configure it like
```java
@Bean(destroyMethod="shutdown")
public SqsConsumer ...
```

Reply:
can you further clarify the difference for my benefit? not sure I understand @derelbenkoenig

Reply:
The code in which we are using the library is using Spring to create the components. The library itself does not know or care about this. So defining the shutdown method of this component is information that is important to the thing that manages the application context (the dependency injection framework), and not to the library code

Reply:
PreDestroy is a javax standard.  It is ignored if you aren't using some sort of mechanism that acknowledges it.  This is the power and the purpose of aspect oriented programming.

Reply:
Sounds good to me

Reply:
Notice that the attribute comes from javax and NOT from Spring

Reply:
Yeah I was aware of that. I was just thinking that the library should be agnostic of how it's used

Reply:
That's why the attribute is a standard java annotation now and not spring specific.  It IS agnostic.

Reply:
That's why I approved the PR :+1:

Reply:
My man. 👍